vesta/services/clash_of_code_entities.py

```diff
@@ -0,0 +1,213 @@
+import datetime
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Optional
+
+import discord
+from discord import Embed
+
+
+class GameMode(Enum):
+    """
+    Represents a Clash of Code game mode
+    """
+
+    FASTEST = 0
+    REVERSE = 1
+    SHORTEST = 2
+
+    def __repr__(self):
+        return f"coc_mode_{self.name.lower()}"
+
+class Role(Enum):
+    """
+    Represents a Clash of Code player role
+    """
+
+    OWNER = 0
+    STANDARD = 1
+
+class State(Enum):
+    """
+    Represents a Clash of Code game state
+    """
+
+    PENDING = 0
+    RUNNING = 1
+    FINISHED = 2
+
+    def __repr__(self):
+        """
+        Gives the translation key for the state
+        """
+        return f"coc_game_state_{self.name.lower()}"
+
+    def __str__(self):
+        """
+        Gives the emoji for the state
+        """
+        return {
+            State.PENDING: "🕒",
+            State.RUNNING: "🔥",
+            State.FINISHED: "🏁"
+        }[self]
+
+@dataclass
+class ClashOfCodePlayer:
+    """
+    Represents a player in a Clash of Code game
+    """
+
+    name: str
+    role: Role
+    rank: int
+    state: State
+
+    def __init__(self, **kwargs):
+        """
+        Initializes the object with the given data
+
+        :param kwargs: The data to initialize the object with
+        :see hydrate
+        """
+        self.hydrate(**kwargs)
```

Comment:
As I told you, it would have probably been better to just use get(dict, "key", default) in this situation

Reply:
You didn't tell me, can you be clearer here please